sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs

```diff
@@ -15,56 +15,31 @@ public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel i
-                          Deserialize(AssertNotNull(user.Cloud, nameof(user.Cloud), nameof(MicrosoftTeamsUserIdentifierModel))),
-                          rawId);
-                }
-
-                return new UnknownIdentifier(rawId);
-            }
+            var kind = identifier.Kind ?? GetKind(identifier);
 
-            if (identifier.CommunicationUser is CommunicationUserIdentifierModel communicationUser)
+            if (kind == CommunicationIdentifierModelKind.CommunicationUser
+                && identifier.CommunicationUser is not null)
             {
-                return new CommunicationUserIdentifier(
-                    AssertNotNull(communicationUser.Id, nameof(communicationUser.Id), nameof(CommunicationUserIdentifierModel)));
+                return new CommunicationUserIdentifier(AssertNotNull(identifier.CommunicationUser.Id, nameof(identifier.CommunicationUser.Id), nameof(CommunicationUserIdentifierModel)));
             }
 
-            if (identifier.PhoneNumber is PhoneNumberIdentifierModel phoneNumber)
+            if (kind == CommunicationIdentifierModelKind.PhoneNumber
+                && identifier.PhoneNumber is not null)
             {
                 return new PhoneNumberIdentifier(
-                    AssertNotNull(phoneNumber.Value, nameof(phoneNumber.Value), nameof(PhoneNumberIdentifierModel)),
+                    AssertNotNull(identifier.PhoneNumber.Value, nameof(identifier.PhoneNumber.Value), nameof(PhoneNumberIdentifierModel)),
                     AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(PhoneNumberIdentifierModel)));
             }
 
-            if (identifier.MicrosoftTeamsUser is MicrosoftTeamsUserIdentifierModel teamsUser)
+            if (kind == CommunicationIdentifierModelKind.MicrosoftTeamsUser
+                && identifier.MicrosoftTeamsUser is not null)
             {
+                var user = identifier.MicrosoftTeamsUser;
                 return new MicrosoftTeamsUserIdentifier(
-                    AssertNotNull(teamsUser.UserId, nameof(teamsUser.UserId), nameof(MicrosoftTeamsUserIdentifierModel)),
-                    AssertNotNull(teamsUser.IsAnonymous, nameof(teamsUser.IsAnonymous), nameof(MicrosoftTeamsUserIdentifierModel)),
-                    Deserialize(AssertNotNull(teamsUser.Cloud, nameof(teamsUser.Cloud), nameof(MicrosoftTeamsUserIdentifierModel))),
-                    rawId);
+                      AssertNotNull(user.UserId, nameof(user.UserId), nameof(MicrosoftTeamsUserIdentifierModel)),
+                      AssertNotNull(user.IsAnonymous, nameof(user.IsAnonymous), nameof(MicrosoftTeamsUserIdentifierModel)),
+                      Deserialize(AssertNotNull(user.Cloud, nameof(user.Cloud), nameof(MicrosoftTeamsUserIdentifierModel))),
+                      rawId);
             }
 
             return new UnknownIdentifier(rawId);
```

Comment:
for other platforms(java&js) we are checking for the non-nullability of rawId first, we have to unify this logic across all platforms

Reply:
@petrsvihlik @AikoBB here RawId is not used for creating `MicrosoftTeamsUserIdentifier` and `CommunicationUserIdentifier` checking it for null before trying to create any of these identifiers will break the existing logic

Reply:
@AikoBB  I've re-checked: there is already a validation of the rawId on the top of the method
https://github.com/Azure/azure-sdk-for-net/blob/8ce74c287bc545f1ea518a3364fe174d3d94e431/sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs#L14
            `string rawId = AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(CommunicationIdentifierModel));`